### DISCUSSION TITLE
Refer to spaces that have apps inside of them

### DISCUSSION THREAD

**Discussion by itaysharir** [FelixKratz/SketchyBar#248] [Q&A] [answered]
I want to make the spaces that have apps running in them a different color then empty spaces, Is there a way to do that?
Thanks in advance!

**Comment by FelixKratz** [comment, marked as answer]
You can do so with yabai, assuming you use the default config, just replace the space.sh plugin script with this one:
```bash
#!/usr/bin/env sh

WIN=$(yabai -m query --spaces --space $SID | jq '.windows[0]')
HAS_WINDOWS="true"
if [ "$WIN" = "null" ];then
  HAS_WINDOWS="false"
fi
sketchybar --set $NAME background.drawing=$SELECTED icon.highlight=$HAS_WINDOWS
```
and define a `icon.highlight_color` for all spaces.

Caveats:
This only updates once you enter or leave a certain space, if you want all spaces to update every time you would need to implement the spaces logic via a regular item with an attached script, subscribed to the relevant events.

**Comment by itaysharir** [comment]
Thanks! One more question though, can i highlight the spaces which have apps in them and also the on I'm in right now? Even if it is empty? I tried adding _icon.highlight=$SELECTED_ to the last line in the script you sent (I created that variable), but then it JUST highlighted the selected space.

**Reply by FelixKratz** [reply]
What you want is to add a conditional for the if statement:
```bash
#!/usr/bin/env sh

WIN=$(yabai -m query --spaces --space $SID | jq '.windows[0]')
HAS_WINDOWS_OR_IS_SELECTED="true"
if [ "$WIN" = "null" ] && [ "$SELECTED" = "false" ];then
  HAS_WINDOWS_OR_IS_SELECTED="false"
fi
sketchybar --set $NAME background.drawing=$SELECTED icon.highlight=$HAS_WINDOWS_OR_IS_SELECTED
```

**Reply by itaysharir** [reply]
Thanks! It worked!
